pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1220,6 +1220,8 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 				if err != nil {
 					return err
 				}
+			} else if iface.Macvtap != nil {
+				domainIface.Type = "ethernet"
```

Comment:
See the code we have for `bridge` and `masquerade`. We should have a big part of it for `macvtap` as well.
- boot order is useful for macvtap as well
- since we use macvtap iface provided by CNI, we also need the prefix handling (multus vs genie)
- the mentioned section already has handling for `source`, how comes it is missing here?

Reply:
Because source is not needed. It would indicate which is the parent iface of the macvtap, which, since the device is *already* created by CNI, is totally irrelevant at this point.

Reply:
Missing:
  - boot order

I don't know if we can support genie; we rely on multus adding the 'deviceID' parameter, so that CNI can eventually import the macvtap interface. 

We'll need to check if genie can do this.

Reply:
I would not bother and document it's unsupport of macvtap.

Reply:
Implemented bootOrder.